vpr/src/place/initial_placement.cpp

```diff
@@ -298,11 +331,33 @@ static void initial_placement_blocks(int* free_locations, enum e_pad_loc_type pa
                             cluster_ctx.clb_nlist.block_name(blk_id).c_str(), size_t(blk_id), logical_block->name, logical_block->index);
         }
 
-        itype = type->index;
+        int itype = type->index;
+
+        auto possible_sub_tiles = get_possible_sub_tile_indices(type, logical_block);
+
+        int isub_tile = OPEN;
+        for (int sub_tile : possible_sub_tiles) {
+            if (free_locations[itype][sub_tile] > 0) {
+                isub_tile = sub_tile;
+                break;
+            }
+        }
```

Comment:
Need comments here, or perhaps make into a named sub-routine (e.g. find_empty_subtile())?

Reply:
Done (called the sub routing get_free_subtile)